gateway/handlers/logs.go

```diff
@@ -0,0 +1,225 @@
+package handlers
+
+import (
+	"context"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+)
+
+const crlf = "\r\n"
+const upstreamLogsEndpoint = "/system/logs"
+
+// NewLogHandlerFunc creates and http HandlerFunc from the supplied log Requestor.
+func NewLogHandlerFunc(logProvider url.URL) http.HandlerFunc {
+	writeRequestURI := false
```

Comment:
Initial version - bring in a time.Duration and kill the connection using `time.AfterFunc()`

Reply:
This was implemented with timeout, as in the the next comment